inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -501,12 +501,23 @@ void GNAGraphCompiler::PowerPrimitive(InferenceEngine::CNNLayerPtr layer) {
     auto input = layer->insData[0].lock();
 
     auto outputs = *layer->outData.begin();
-
-    uint32_t num_rows_in = FROM_IR_DIM(input, 1);
-    uint32_t num_columns_in = FROM_IR_DIM(input, 2);
+    uint32_t num_rows_in = InferenceEngine::details::product(begin(input->getDims()), end(input->getDims()));
+    uint32_t num_columns_in = 1;
     uint32_t num_rows_out = num_rows_in;
     uint32_t num_padding = ALIGN(num_rows_in, 8) - num_rows_in;
 
+    if (input->getDims().size() > 2 || input->getDims()[0] >= 8) {
+        for (size_t index_divide = 8; index_divide > 0; index_divide--) {
```

Comment:
loop can be simplified: 
1. index can be moved output of loop scope
2. stop condition changed to index > 1
3. num_rows num_columns calculation can be done out of loop. 
Otherwise we looks testing against division by 1 and taking modulus of this division which seems odd, while it doesn't affect prformance or accuracy, so i leave this up to you